lexicon/tests/providers/test_njalla.py

```diff
@@ -0,0 +1,22 @@
+"""Integration tests for Njalla provider"""
+from unittest import TestCase
+
+import pytest
+from lexicon.tests.providers.integration_tests import IntegrationTests
+
+# Hook into testing framework by inheriting unittest.TestCase and reuse
+# the tests which *each and every* implementation of the interface must
+# pass, by inheritance from integration_tests.IntegrationTests
+
+class NjallaProviderTests(TestCase, IntegrationTests):
+    """TestCase for Njalla"""
+    provider_name = 'njalla'
+    domain = 'example.com'
+
+    def _filter_headers(self):
+        return ['Authorization']
+
+    @pytest.fixture(autouse=True)
+    def _skip_suite(self, request):  # pylint: disable=no-self-use
```

Comment:
Are you willing to implement the advanced test suite? It is mainly about the capacity to handle multiple records for the same name (like TXTs). It is mostly useful for ACME clients, like Certbot, to issue wildcard certificates, which require two token for the same name.

It is not mandatory, but I wanted to give you an insight about it.

Reply:
I ran into an issue with the very first one: `test_provider_when_calling_create_record_with_duplicate_records_should_be_noop`
- This is not the native behavior of Njalla (it allows duplicate records)
- I attempted doing the check myself with a `_list_records` in the beginning of `_create_record`, but it appeared to not return results for the duplicate record created only a moment before. It seems operations aren't synchronous, and it takes a while for the system to become consistent.

As completely duplicate records are permitted, record sets are also, and are returned correctly by `_list_records` with filter parameters.

I have however not implemented `def test_provider_when_calling_delete_record_with_record_set_name_remove_all`. Is deleting record sets in one go required for by certbot?

Reply:
Alright, may be tricky to match the expected behavior for the advanced test suite. No, deleting all the record set is not mandatory: Certbot will remove a specific TXT+entry+value for each related challenge in a given certificate.